pkg/controller/deliverable/reconciler.go

```diff
@@ -41,21 +42,24 @@ type Reconciler struct {
 	Realizer                realizer.Realizer
 	DynamicTracker          tracker.DynamicTracker
 	conditionManager        conditions.ConditionManager
-	logger                  logr.Logger
 }
 
 func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	r.logger = logr.FromContext(ctx)
-	r.logger.Info("started")
-	defer r.logger.Info("finished")
+	log := logr.FromContext(ctx)
+	log.Info("started")
+	defer log.Info("finished")
+
+	log = log.WithValues("deliverable", req.NamespacedName)
```

Comment:
do we need this? controller-runtime is already adding `name` and `namespace` to the object

e.g., with this approach, we have:

```
{
  "level": "info",
  "ts": 1637596812.8353918,
  "logger": "controller.delivery",
  "msg": "started",
  "name": "delivery",
  "namespace": ""
}

{
  "level": "info",
  "ts": 1637596812.8363948,
  "logger": "controller.deliverable",
  "msg": "started",
  "name": "test",
  "namespace": "default"
}

{
  "level": "info",
  "ts": 1637596812.8366206,
  "logger": "controller.deliverable",
  "msg": "delivery is not in ready state",
  "name": "test",
  "namespace": "default",
  "deliverable": "default/test",
  "delivery": "delivery"
}
```

Reply:
Our thoughts here were to match log messages with the context eg:

`log.Info("My deliverable is unhappy about widget", "widget", widget)`

The guidance from controller runtime is to not `fmt`  the message, but reference the keys you will find in the structure log. So mixed with the context, the above log line produces:

```
{
  "msg": "My deliverable is unhappy about widget",
  "deliverable": "default/my-deliverable",
  "widget": {
    "apiVersion": "hello.io/widgetctrlr",
    "kind": "Widget",
    "namespace": "my-ns",
    "name": "my-widget"
  }
}
```

Reply:
I mean, controller-runtime by default will inject namespace and name, won't it? 🤔  my point was more that we're duplicating that information

Reply:
Yes but it reads like this:

Reply:
aaaaah, I see 👍 

imo there's still duplication though - `controller.deliverable` already lets you know it's a deliverable